src/signup.js

```diff
@@ -1,131 +1,176 @@
+      findClassNameErrorMessage.remove();
       passInput.style.borderColor = '#CCD5E3';
     }
   }
-}
+})
 
 //비밀번호 확인 에러메세지
-function passCheckErrorMessage (e) {
-  //에러메세지 - 값이 다를 경우
-  if (passInput.value !== passCheckInput.value) {
-    errorMessage(passCheckInput, '비밀번호가 일치하지 않습니다.');
+const passCheckInput = document.querySelector('#passCheck');
+passCheckInput.addEventListener('focusout', (e) => {
+  const passwordCheck = passInput.value === passCheckInput.value;
+  const passCheckEmpty = passCheckInput.value.trim() === '';
+  const findClassNameErrorMessage = passCheckInput.parentElement.querySelector('.errorMessage');
+  
+  if (!passwordCheck) {
+    setErrorMessageToElement (passCheckInput, '비밀번호가 일치하지 않습니다.');
+  } else if (passCheckEmpty) {
+    setErrorMessageToElement (passCheckInput, '비밀번호를 입력해주세요.');
   } else {
-    if (passCheckInput.parentElement.parentElement.querySelector('.errorMessage')) {
-      passCheckInput.parentElement.parentElement.querySelector('.errorMessage').remove();
+    if (findClassNameErrorMessage) {
+      findClassNameErrorMessage.remove();
       passCheckInput.style.borderColor = '#CCD5E3';
     }
   }
-}
+});
 
-//로그인 시도
-function submitForm (e) {
-  //현재페이지에서 테스트
+
+//눈 모양 클릭
+const toggle = document.querySelectorAll ('.fa-solid');
+toggle.forEach (toggle => {
+  toggle.addEventListener('click', (e) => {
+    const toggleIcon = toggle.previousElementSibling;
+    
+    if (toggleIcon.type === 'password') {
+      toggleIcon.type = 'text';
+      toggle.classList.remove('fa-eye-slash');
+      toggle.classList.add('fa-eye');
+    } else {
+      toggleIcon.type = 'password';
+      toggle.classList.remove('fa-eye');
+      toggle.classList.add('fa-eye-slash');
+    }
+  })
+});
+
+//회원가입 시도
+// signForm.addEventListener('submit', (e) => {
+//   const validateSignUp = validateEmail(emailInput.value) && validatePass(passInput.value);
+//   //현재페이지에서 테스트
+//   e.preventDefault();
+  
+//   if (validateSignUp) {
+//     window.location.href = "./folder";
+//   } else {
+//     setErrorMessageToElement (emailInput, '이메일을 확인해주세요.');
+//     setErrorMessageToElement (passInput, '비밀번호를 확인해주세요.');
+//     setErrorMessageToElement (passCheckInput, '비밀번호를 확인해주세요.');
+//   }
+// });
+
+const signForm = document.querySelector('.signup');
+signForm.addEventListener('submit', async (e) => {
   e.preventDefault();
+  const userEmail = emailInput.value;
 
-  if (emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101') {
-    window.location.href = "./folder";
-  } else {
-    errorMessage(emailInput, '이메일을 확인해주세요.');
-    errorMessage(passInput, '비밀번호를 확인해주세요.');
-    errorMessage(passCheckInput, '비밀번호를 확인해주세요.')
-  }
-}
+  // 중복확인
+  try {
+    const response = await fetch('https://bootcamp-api.codeit.kr/api/check-email', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ userEmail }),
```

Comment:
docs문서를 확인 해보시면 body에 어떤 데이터 타입이 들어가야하는지 알 수 있습니다.

```suggestion
      body: JSON.stringify({ email:userEmail }),
```